packages/core/src/carousel/private/utils/useInterval.ts

```diff
@@ -1,7 +1,12 @@
-import { useEffect, useRef } from "react";
+import { useEffect, useRef, useState } from "react";
 
-const useInterval = (callback: () => void, delay: number) => {
+const useInterval = (
+  callback: () => void,
+  delay: number,
+  isDragging?: boolean,
+) => {
   const savedCallback = useRef<() => void>();
+  const [intervalId, setIntervalId] = useState<number | undefined>();
```

Comment:
intervalId state may potentially be avoided. Does it make any sense to use ref instead?

Reply:
It makes total sense. Updated.